### DISCUSSION TITLE
monai.transforms.RandRotated

### DISCUSSION THREAD

**Discussion by QuyLe-Minh** [Project-MONAI/tutorials#1647] [Q&A] [answered]
Hi,
I'm new to monai framework. Currently, I'm doing research related to liver tumor segmentation and I want to augment my dataset. One of modules I used is RandRotated and the instructions are quite unclear for me:

> range_x – Range of rotation angle in radians in the plane defined by the first and second axes. If single number, angle is uniformly sampled from (-range_x, range_x).
> 
> range_y – Range of rotation angle in radians in the plane defined by the first and third axes. If single number, angle is uniformly sampled from (-range_y, range_y). only work for 3D data.
> 
> range_z – Range of rotation angle in radians in the plane defined by the second and third axes. If single number, angle is uniformly sampled from (-range_z, range_z). only work for 3D data.

I wonder what are the second and third axes defined in range_z (for example). Since I fed in data of shape (C, D, H, W), I got undesired behavior such that a slice is not rotated when the second and third axes of my cube are H,W. But, when the data has a shape of (C, H, W, D), rotation seems correctly.

Thanks for spending time helping me out.
Have a nice day.

**Comment by KumoLiu** [comment]
Hi @QuyLe-Minh, thanks for your interest here.
I don't think it's relevant to how you define the data, it's just the plane of the first and third axes (e.g. the x-z plane).

Hope it helps, thanks.

**Reply by QuyLe-Minh** [reply]
Yes, my ultimate question is that what are x and z plane actually? Because my data (3Dircadb for e.g) is a stack of bunch of slices, if i want to rotate slices 20 degrees, I must rotate plane of the 2nd and 3rd axes of data of shape (C,H,W,D).

**Reply by KumoLiu** [reply, marked as answer]
When spatial_dims == 3, the `range x y z` corresponds to rotation in the 1st, 2nd, and 3rd dim respectively.
